### PR TITLE
Convert templates to mscz at build time

### PR DESCRIPTION
**Pull request by dmitrio95**
This pull request contains CMake scripts changes necessary to convert .mscx templates to .mscz file format during the build time. This makes it faster for MuseScore to generate thumbnails for templates which speeds up the new score dialog creation, also it makes the templates directory somewhat less in size (by ≈1.9 MB in my tests).

This pull request makes two assumptions:
1) In order not to generate templates at each rebuild, it does not make each conversion job depend on MuseScore executable directly, instead the whole templates generation target depends on it. Thus the assumption is made that MuseScore is unlikely to change in a way that significantly changes the result of templates conversion to mscz format, if that is not true you would need to trigger templates regeneration manually.
2) As this templates conversion takes place at build time, MuseScore may not find installed files and complains (for debug build) of not finding `styles/chords_std.xml`. It is probably not needed as long as standard templates do not contain chord symbols but if such templates appear it would be necessary to make it possible for MuseScore to find that file in the source tree or in the build directory.
 
Also this PR removes `convert.json` batch file for templates conversion added in efbdbffa7b6c42feae607aca8ec236ddcda18946. That file does not seem to be used by any build scripts, and the information in it is a duplicate of that in the corresponding CMakeLists.txt. Feel free to correct me if it is necessary for some purposes and it is better to return it back.

### CODE REVIEW

File: share/templates/CMakeLists.txt [hunk 1]
```diff
@@ -17,20 +17,53 @@
...
 
+# Usage: install_templates(install_dir [mscx_templates...])
+# MSCX templates paths should be relative to CMAKE_CURRENT_SOURCE_DIR.
+function(install_templates DESTINATION_DIR)
+      # Create a directory for the converted templates
+      get_filename_component(MSCZ_BUILD_DIR "${ARGV1}" DIRECTORY)
+      file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${MSCZ_BUILD_DIR}")
+
+      # Remove DESTINATION_DIR from the list
+      list(REMOVE_AT ARGV 0)
```
Comment (shoogle) [inline share/templates/CMakeLists.txt]: FYI there is another list variable `ARGN` that will have DESTINATION_DIR already removed. See [CMake function](https://cmake.org/cmake/help/v3.0/command/function.html).

Comment (dmitrio95) [reply share/templates/CMakeLists.txt]: I didn't notice the note about `ARGN` there, thanks!

**Comment by dmitrio95** [comment]
It seems that this conversion wouldn't be possible on systems without graphical capabilities since MuseScore is not able to run in a truly non-GUI mode. So either the whole build process should be launched under some graphical environment emulation tools on such systems, like it is done for tests (see [this](https://github.com/musescore/MuseScore/blob/344b79220c564bf52e057d8e3d41bbddf8a413cb/build/travis/job1_Tests/run_tests.sh#L8-L10)), or the conversion process should be organized differently.

**Comment by Jojo-Schmitz** [comment]
Ishn't there a Virtual Frame Buffer X-driver?
Ah yes, `xvfb-run`...

**Pull request closed without merging**